web/src/pages/Route/components/Step1/MetaView.tsx

```diff
@@ -258,67 +269,195 @@ const MetaView: React.FC<RouteModule.Step1PassProps> = ({
     );
   };
 
-  const CustomRedirect: React.FC = () => (
-    <Form.Item
-      noStyle
-      shouldUpdate={(prev, next) => {
-        if (prev.redirectOption !== next.redirectOption) {
-          onChange({ action: 'redirectOptionChange', data: next.redirectOption });
-        }
-        return prev.redirectOption !== next.redirectOption;
-      }}
-    >
-      {() => {
-        if (form.getFieldValue('redirectOption') === 'customRedirect') {
-          return (
+  const CustomRedirectType = () => {
+    switch (form.getFieldValue('customRedirectType')) {
+      case CUSTOM_REDIRECT_TYPE.STATIC:
+        return (
+          <Form.Item
+            label={formatMessage({ id: 'page.route.form.itemLabel.redirectCustom' })}
+            required
+            style={{ marginBottom: 0 }}
+          >
+            <Row gutter={10}>
+              <Col span={5}>
+                <Form.Item
+                  name="redirectURI"
+                  rules={[
+                    {
+                      required: true,
+                      message: `${formatMessage({
+                        id: 'component.global.pleaseEnter',
+                      })}${formatMessage({
+                        id: 'page.route.form.itemLabel.redirectURI',
+                      })}`,
```

Comment:
It is better not to use template strings to complete the internationalization, the Chinese state will have extra spaces